src/main/java/com/elytradev/concrete/block/ConstructionBehaviour.java

```diff
@@ -0,0 +1,83 @@
+package com.elytradev.concrete.block;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.Item;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import java.util.function.Supplier;
+
+/**
+ * An interface used to define the behaviour that will be used to construct a {@link ConcreteBlock},
+ * from the Concrete block builder system.
+ *
+ * This allows for an extended {@link ConcreteBlock} in special cases where such functionality may
+ * be desired, such as glass.
+ */
+public interface ConstructionBehaviour {
```

Comment:
I don't like the idea of this class to begin with. We're taking concerns that are currently unecessarily tightly coupled with Block, and unnecessarily tightly coupling them to ConstructionBehaviour (in your other code you had mentioned lumping "ore" in here, which suggested to me this was a busted concept). Can we take each of these concerns, and make each one their own "idea"? Glass, for instance, is translucent. So let's call it what it is, translucency, and make it its own concern.